yapapi/engine.py

```diff
@@ -107,6 +107,8 @@ def __init__(
         :param budget: maximum budget for payments
         :param strategy: market strategy used to select providers from the market
             (e.g. LeastExpensiveLinearPayuMS or DummyMS)
+        :param event_consumer: callable that will be directly executed on every Event this Engine creates.
+            NOTE: it is expected to be fast - if not, it will block the _Engine.
```

Comment:
it's not immediately clear that this is the reason that we're implementing it as an AsyncWrapper

Reply:
Actually, `_Engine` knows nothing about the existence of `AsyncWrapper`s and/or `AsyncEventDispatcher`, so I'm not sure if this should be mentioned here. Nevertheless, if you think this comment is misleading, please propose an improvement :)

Reply:
maybe instead of fast, just say "non-blocking" ...

```suggestion
            NOTE: it is expected to be fast or async - if not, it will block the _Engine.
```